src/app/main/index.js

```diff
@@ -145,67 +145,9 @@ angular
     });
 
     $scope.$watch('search.query', function(q) {
-        $scope.search.results = assignSearchRelevance(projectService.search(q));
+        $scope.search.results = projectService.search(q);
```

Comment:
I think that both this line and this line in `components/search/search.js` are both called when the `search.query` variable changes. I'm seeing some input lag between typing a key and seeing results (~150 models in my test project) and I wonder if running the search twice for each keystroke contributes to the latency?
```
scope.results = filterResults(projectService.search(scope.query), scope.checkboxStatus);
```

Reply:
I made a quick loom with some of my observations & open questions - check it out here! https://www.loom.com/share/8422b5b0865d40e0a477cf25cb0815e5

Reply:
I've taken the redundant search call out of index.js - on my computer it's still sitting around 350ms per keystroke which is odd, I would have expected that to be much lower now that it's only doing the work once. I also did some basic Googling on $$debounceViewValueCommit but nothing jumped right out at me :(